File: need4spd/euler_30.rb
limit=(9**5)*6

r_list=Array.new

(2..limit).each do
  |n|
  n_str = n.to_s
  
  sum=0
  n_str.each_char do
    |c| sum = sum + (c.to_i ** 5)
  end
  
  if (sum==n)
    r_list.push(n)
  end
end

puts r_list.inject(:+)

File: need4spd/euler_30.py

r_list=list()

exponent=5

limit=(9**5)*6

for n in range(2, limit+1):
    n_str = str(n)
    
    if n == sum(int(i) ** 5 for i in n_str):
        r_list.append(n)
        
print(sum(r_list))